### PR TITLE
Add a POST API method for habiticaRegister

### PR DESCRIPTION
**Pull request by shankari**
This is an exemplar to show how to add an externally invocable POST API method
to the e-mission server.

For further explanation and examples, please consult the documentation for bottle.
http://bottlepy.org/docs/dev/index.html

@juemura, @sunil07t, @yw374cornell: this might be helpful to you as you
consider adding other methods.

This method would be invoked by sending a HTTP POST

`$http.post()`
or
`requests.post()`

with data {'username': ..., 'email': ....}

While calling this from the emission app, you should use a helper function
defined in `CommHelper` so that the JWT is inserted correctly into the HTTP
POST message.

### CODE REVIEW

**Comment by shankari** [comment]
I felt this was faster than yet another round of explaining with text snippets or diagrams.
THIS HAS NOT BEEN TESTED.
But I expect it to work with minor bug fixes.

**Comment by shankari** [comment]
jenkins, test this please

**Comment by shankari** [comment]
jenkins, test this please

File: emission/net/api/cfc_webapp.py [hunk 1]
```diff
@@ -435,6 +436,21 @@ def movesCallback():
...
   state = request.json['state']
   return auth.movesCallback(code, state, user_uuid)
+
+@post('/habiticaRegister')
+def habiticaRegister():
+  logging.debug("habitica registration request %s from user = %s" %
+                (request.json, request))
+  user_uuid = getUUID(request)
+  assert(user_uuid is not None)
+  user = User.fromUUID(user_uuid)
```
Comment (juemura) [inline emission/net/api/cfc_webapp.py]: why do you create a user here? I don't see where it's used.

Comment (shankari) [inline emission/net/api/cfc_webapp.py]: Not used yet, but will be used soon to get the stored email address.

**Comment by shankari** [comment]
Changed the expected format slightly to match the authenticated API call from the phone and to be a better example for future changes.
phone service: https://github.com/e-mission/e-mission-phone/pull/63
calls comm function: https://github.com/e-mission/cordova-server-communication/pull/12

**Pull request merged**